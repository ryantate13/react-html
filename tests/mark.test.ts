import {mark} from '../src';

describe('mark', () => {
    it('renders with children', () => {
        expect(mark({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(mark({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(mark()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(mark({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(mark({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(mark({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(mark({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(mark({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(mark({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(mark({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(mark({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(mark({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(mark({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(mark({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(mark({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(mark({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(mark({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(mark({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(mark({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(mark({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(mark({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(mark({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(mark({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(mark({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(mark({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(mark({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(mark({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(mark({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(mark({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(mark({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(mark({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(mark({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(mark({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(mark({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(mark({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(mark({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(mark({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(mark({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(mark({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(mark({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(mark({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(mark({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(mark({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(mark({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(mark({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(mark({translate: "no"})).toMatchSnapshot();
    });
});