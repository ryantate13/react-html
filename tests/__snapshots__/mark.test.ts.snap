// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mark renders arrays of children 1`] = `
<mark>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</mark>
`;

exports[`mark renders with accessKey="a,b,c,d" 1`] = `
<mark
  accessKey="a,b,c,d"
/>
`;

exports[`mark renders with autoCapitalize="characters" 1`] = `
<mark
  autoCapitalize="characters"
/>
`;

exports[`mark renders with autoCapitalize="none" 1`] = `
<mark
  autoCapitalize="none"
/>
`;

exports[`mark renders with autoCapitalize="off" 1`] = `
<mark
  autoCapitalize="off"
/>
`;

exports[`mark renders with autoCapitalize="on" 1`] = `
<mark
  autoCapitalize="on"
/>
`;

exports[`mark renders with autoCapitalize="sentences" 1`] = `
<mark
  autoCapitalize="sentences"
/>
`;

exports[`mark renders with autoCapitalize="words" 1`] = `
<mark
  autoCapitalize="words"
/>
`;

exports[`mark renders with children 1`] = `
<mark>
  TEST
</mark>
`;

exports[`mark renders with className="foo bar baz" 1`] = `
<mark
  className="foo bar baz"
/>
`;

exports[`mark renders with contentEditable="false" 1`] = `
<mark
  contentEditable="false"
/>
`;

exports[`mark renders with contentEditable="true" 1`] = `
<mark
  contentEditable="true"
/>
`;

exports[`mark renders with dir="auto" 1`] = `
<mark
  dir="auto"
/>
`;

exports[`mark renders with dir="ltr" 1`] = `
<mark
  dir="ltr"
/>
`;

exports[`mark renders with dir="rtl" 1`] = `
<mark
  dir="rtl"
/>
`;

exports[`mark renders with draggable="false" 1`] = `
<mark
  draggable="false"
/>
`;

exports[`mark renders with draggable="true" 1`] = `
<mark
  draggable="true"
/>
`;

exports[`mark renders with empty attributes 1`] = `<mark />`;

exports[`mark renders with hidden=false 1`] = `
<mark
  hidden={false}
/>
`;

exports[`mark renders with hidden=true 1`] = `
<mark
  hidden={true}
/>
`;

exports[`mark renders with id="TEST" 1`] = `
<mark
  id="TEST"
/>
`;

exports[`mark renders with inputMode="decimal" 1`] = `
<mark
  inputMode="decimal"
/>
`;

exports[`mark renders with inputMode="email" 1`] = `
<mark
  inputMode="email"
/>
`;

exports[`mark renders with inputMode="none" 1`] = `
<mark
  inputMode="none"
/>
`;

exports[`mark renders with inputMode="numeric" 1`] = `
<mark
  inputMode="numeric"
/>
`;

exports[`mark renders with inputMode="search" 1`] = `
<mark
  inputMode="search"
/>
`;

exports[`mark renders with inputMode="tel" 1`] = `
<mark
  inputMode="tel"
/>
`;

exports[`mark renders with inputMode="text" 1`] = `
<mark
  inputMode="text"
/>
`;

exports[`mark renders with inputMode="url" 1`] = `
<mark
  inputMode="url"
/>
`;

exports[`mark renders with is="foo" 1`] = `
<mark
  is="foo"
/>
`;

exports[`mark renders with itemID="https://google.com/" 1`] = `
<mark
  itemID="https://google.com/"
/>
`;

exports[`mark renders with itemProp="https://google.com/" 1`] = `
<mark
  itemProp="https://google.com/"
/>
`;

exports[`mark renders with itemRef="foo bar" 1`] = `
<mark
  itemRef="foo bar"
/>
`;

exports[`mark renders with itemScope=false 1`] = `
<mark
  itemScope={false}
/>
`;

exports[`mark renders with itemScope=true 1`] = `
<mark
  itemScope={true}
/>
`;

exports[`mark renders with itemType="https://google.com/" 1`] = `
<mark
  itemType="https://google.com/"
/>
`;

exports[`mark renders with lang="" 1`] = `
<mark
  lang=""
/>
`;

exports[`mark renders with lang="es-419" 1`] = `
<mark
  lang="es-419"
/>
`;

exports[`mark renders with slot="TEST" 1`] = `
<mark
  slot="TEST"
/>
`;

exports[`mark renders with spellCheck="false" 1`] = `
<mark
  spellCheck="false"
/>
`;

exports[`mark renders with spellCheck="true" 1`] = `
<mark
  spellCheck="true"
/>
`;

exports[`mark renders with style={"color":"#000000"} 1`] = `
<mark
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`mark renders with tabIndex=1 1`] = `
<mark
  tabIndex={1}
/>
`;

exports[`mark renders with title="TEST" 1`] = `
<mark
  title="TEST"
/>
`;

exports[`mark renders with translate="no" 1`] = `
<mark
  translate="no"
/>
`;

exports[`mark renders with translate="yes" 1`] = `
<mark
  translate="yes"
/>
`;

exports[`mark renders without attributes 1`] = `<mark />`;
